hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/DefaultRecordStore.java

```diff
@@ -765,6 +774,7 @@ protected Object putInternal(Data key, Object value, long ttl, boolean loadFromS
         if (record == null) {
             record = createRecord(value, ttl, now);
             storage.put(key, record);
+            eventJournal.writeAddEvent(mapContainer.getObjectNamespace(), partitionId, record.getKey(), record.getValue());
```

Comment:
I don't think we will be adding the store any time soon - it can severely slow down the partition operations, even to the point of stalling them. We would also need some way of determining if it is a backup operation so it doesn't get stored twice. 
Currently it is a simple "put in array" which can hardly fail. If and when we will add the ringbuffer store to the event journal we can easily switch places for all calls to the record store and event journal.